scalableminds/webknossos
# Shrink viewport gaps

This PR makes the layout gaps a bit more dense and also fixes #3249. Unfortunately, I didn't manage to fix the later issue in a nice manner. The current code is not very elegant, since the canvas size depends on the available size which in turn depends on the layout scale. However, we only listen to the layout scale and then read the available size (which does not reflect the layout scale, initially). However, I think, that the current solution is fine as a hot fix.

### URL of deployed dev instance (used for testing):
- https://shrinkviewportgaps.webknossos.xyz

### Steps to test:
- load layout
- use k to oversize the layout and reload
- the off-screen part of the canvas should still be painted

### Issues:
- fixes #3255 
- fixes #3249

------
